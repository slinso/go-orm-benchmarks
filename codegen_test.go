package main_test

import (
	"database/sql"
	"testing"

	"github.com/efectn/go-orm-benchmarks/bench/codegen"
	"github.com/efectn/go-orm-benchmarks/bench/codegen/db/mysql"
	"github.com/gkampitakis/go-snaps/snaps"
	_ "github.com/jackc/pgx/v4/stdlib"
)

func TestCodegen_ReadSlice(t *testing.T) {
	var err error
	db, err := sql.Open("pgx", "host=localhost user=postgres password=postgres dbname=test sslmode=disable")
	if err != nil {
		t.Fatal(err)
	}

	m := &codegen.Models{}
	ctx := codegen.NewContext()
	store := mysql.NewModelsStore(ctx, db)

	for i := 0; i < 100; i++ {
		m.ID = 0
		err := store.Insert(m)
		if err != nil {
			t.Fatal(err)
		}
	}

	data, err := store.Limit(100).Query()
	if err != nil {
		t.Fatal(err)
	}

	snaps.MatchSnapshot(t, data)
}
