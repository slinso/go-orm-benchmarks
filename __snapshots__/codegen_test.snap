
[TestCodegen_ReadSlice - 1]
[]*codegen.Models{
    &codegen.Models{ID:3, Name:"Orm Benchmark", Title:"Just a Benchmark for fun", Fax:"99909990", Web:"http://blog.milkpod29.me", Age:100, Right:true, Counter:1000},
    &codegen.Models{ID:4, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:5, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:6, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:7, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:8, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:9, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:10, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:11, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:12, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:13, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:14, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:15, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:16, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:17, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:2, Name:"Orm Benchmark", Title:"Just a Benchmark for fun", Fax:"99909990", Web:"http://blog.milkpod29.me", Age:100, Right:true, Counter:1000},
    &codegen.Models{ID:18, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:19, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:20, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:21, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:22, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:23, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:24, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:25, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:26, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:27, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:28, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:29, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:30, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:31, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:32, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:1, Name:"Orm Benchmark", Title:"Just a Benchmark for fun", Fax:"99909990", Web:"http://blog.milkpod29.me", Age:100, Right:true, Counter:1000},
    &codegen.Models{ID:33, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:34, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:35, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:36, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:37, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:38, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:39, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:40, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:41, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:42, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:43, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:44, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:45, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:46, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:47, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:48, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:49, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:50, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:51, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:52, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:53, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:54, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:55, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:56, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:57, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:58, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:59, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:60, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:61, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:62, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:63, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:64, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:65, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:66, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:67, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:68, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:69, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:70, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:71, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:72, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:73, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:74, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:75, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:76, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:77, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:78, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:79, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:80, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:81, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:82, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:83, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:84, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:85, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:86, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:87, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:88, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:89, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:90, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:91, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:92, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:93, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:94, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:95, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:96, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:97, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:98, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:99, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
    &codegen.Models{ID:100, Name:"", Title:"", Fax:"", Web:"", Age:0, Right:false, Counter:0},
}
---
